lib/task_005_lib.rb

```diff
@@ -0,0 +1,95 @@
+module Task005
+  module_function
+
+  class Point
+    attr_reader :x, :y
+
+    def self.from_input(point)
+      x, y = point.split(',').map(&:to_i)
+      Point.new(x, y)
+    end
+
+    def initialize(x, y)
+      @x = x
+      @y = y
+    end
+  end
+
+  class Line
+    attr_reader :point1, :point2
+
+    def self.from_input(line)
+      point1, point2 = line.split(' -> ').map { |point| Point.from_input(point) }
+      Line.new(point1, point2)
+    end
+
+    def initialize(point1, point2)
+      @point1 = point1
+      @point2 = point2
+    end
+
+    def horizontal_or_vertical?(line)
+      line.point1.x == line.point2.x || line.point1.y == line.point2.y
+    end
```

Comment:
Currently, I could do:

```
other_line = Line.new(Point(5,5), Point(5,5))

line = Line.new(Point(0,0), Point(0,1))
line.horizontal_or_vertical?(other_line) # does it make sense?
```

My suggestion:

```suggestion
    def horizontal_or_vertical?
      @point1.x == @point2.x || @point1.y == @point2.y
    end
```

Reply:
The same applies to `degree_45?(line)`

Reply:
And to `points_on_line(line)`

Reply:
:0  yes of course, when moving methods needs to remember that position where method looks from is different